test/xpu_api/utilities/meta/meta.unary/meta.unary.prop/is_nothrow_move_constructible.pass.cpp

```diff
@@ -0,0 +1,110 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+#include "support/utils_invoke.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <class T>
+void
+test_is_nothrow_move_constructible(sycl::queue& deviceQueue)
+{
+    deviceQueue.submit([&](sycl::handler& cgh) {
+        cgh.single_task<T>([=]() {
+            static_assert(dpl::is_nothrow_move_constructible<T>::value);
+            static_assert(dpl::is_nothrow_move_constructible<const T>::value);
+            static_assert(dpl::is_nothrow_move_constructible_v<T>);
+            static_assert(dpl::is_nothrow_move_constructible_v<const T>);
+        });
+    });
+}
+
+template <class T>
+void
+test_has_not_nothrow_move_constructor(sycl::queue& deviceQueue)
+{
+    deviceQueue.submit([&](sycl::handler& cgh) {
+        cgh.single_task<T>([=]() {
+            static_assert(!dpl::is_nothrow_move_constructible<T>::value);
+            static_assert(!dpl::is_nothrow_move_constructible<const T>::value);
+            static_assert(!dpl::is_nothrow_move_constructible<volatile T>::value);
+            static_assert(!dpl::is_nothrow_move_constructible<const volatile T>::value);
+            static_assert(!dpl::is_nothrow_move_constructible_v<T>);
+            static_assert(!dpl::is_nothrow_move_constructible_v<const T>);
+            static_assert(!dpl::is_nothrow_move_constructible_v<volatile T>);
+            static_assert(!dpl::is_nothrow_move_constructible_v<const volatile T>);
+        });
+    });
+}
+
+class Empty
+{
+};
+
+union Union {
+};
+
+struct bit_zero
+{
+    int : 0;
+};
+
+struct A
+{
+    A(const A&);
+};
+
+struct ANT
+{
+    ANT(const ANT&) noexcept;
+};
```

Comment:
From the standard perspective, these structures are move constructible, but we don't test the structures with actual move semantics (taking rvalue reference argument). I propose to change these definitions to take `A&&` and `ANT&&` or add separate structures and test cases.

Reply:
Fixed